src/js/BookReader/BookModel.js

```diff
@@ -272,18 +322,81 @@ export class BookModel {
    * Helper. Return a prop for a given index
    * @param {PageIndex} index
    * @param {keyof PageData} prop
+   * @param {*} def default if property not on the record
    * @return {*}
    */
-  _getDataProp(index, prop) {
+  _getDataProp(index, prop, def=undefined) {
     const dataf = this._getDataFlattened();
     if (isNaN(index) || index < 0 || index >= dataf.length)
       return;
     if ('undefined' == typeof(dataf[index][prop]))
-      return;
+      return def;
     return dataf[index][prop];
   }
 }
 
+/**
+ * A controlled schema for page data.
+ */
+class BookPage {
+  /**
+   * @param {BookModel} book
+   * @param {PageIndex} index
+   */
+  constructor(book, index) {
+    this.book = book;
+    this.index = index;
+    this.width = book.getPageWidth(index);
+    this.height = book.getPageHeight(index);
+    this.pageSide = book.getPageSide(index);
+
+    /** @type {boolean} */
+    this.isViewable = book._getDataProp(index, 'viewable', true);
+    /** @type {PageIndex} The first in the series of unviewable pages this is in. */
+    this.unviewablesStart = book._getDataProp(index, 'unviewablesStart') || null;
+    /**
+     * Consecutive unviewable pages are pages in an unviewable "chunk" which are not the first
+     * of that chunk.
+     */
+    this.isConsecutiveUnviewable = !this.isViewable && this.unviewablesStart != this.index;
+  }
+
+  get prev() {
+    return this.index > 0 ? new BookPage(this.book, this.index - 1) : null;
+  }
+
+  get next() {
+    return this.index < this.book.getNumLeafs() - 1 ? new BookPage(this.book, this.index + 1) : null;
+  }
+
+  get nextCollapsedUnviewables() {
+    if (this.index == this.book.getNumLeafs() - 1) return null;
```

Comment:
suggestion: use `===` 

q: can we make this as part of the constructor since we refer it so much? `his.book.getNumLeafs() - 1` => `this.index`? Or perhaps we can glean it from the incoming page metadata? what do you think?

Reply:
Removed the need for that check altogether when added the `findNext` method 🎉

Reply:
(This is why I really like generators :) )